File: 2017/cs/Day1.cs
namespace AocRunner;

public class Day1
{
    public static void Run(string input, string[] lines)
    {
        int GetSum(string line, int shift) => 
            Enumerable.Range(0, line.Length).Where(i => input[i] == input[(i + shift < input.Length) ? i + shift : i + shift - input.Length])
                                            .Select(i => input[i] - '0')
                                            .Sum();
        
        System.Console.WriteLine($"Part 1: {GetSum(input, 1)}");
        System.Console.WriteLine($"Part 2: {GetSum(input, input.Length / 2)}");
    }
}

File: 2017/haskell/day1.hs
import System.IO

part1 = do
    s <- readFile "../inputs/day1"
    let x = [ read [s !! x] :: Int | x <- [0..(length s - 1)], s !! x == s !! ( if x + 1 > (length s - 1) then x - length s + 1 else x + 1)]
    print (sum x)
    
